canonical-web-and-design/maas-ui
# feat(ui): use action button for delete confirms

## Done

- Migrate TableDeleteConfirm to typescript and update it to use an action button.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Got to settings and preferences.
- Check that the delete confirmation button shows a spinner when deleting items from the tables.

## Fixes

Fixes: #2587.
